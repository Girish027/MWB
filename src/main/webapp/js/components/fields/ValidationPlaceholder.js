import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { AlertIcon } from '@tfs/ui-components';
import { validationErrorStyle } from '../../styles/customComponentsStyles';

export default class ValidationPlaceholder extends PureComponent {
  render() {
    const { validationMessage, styleOverride } = this.props;

    let validationStyle = validationErrorStyle;
    if (styleOverride) {
      validationStyle = { ...validationErrorStyle, ...styleOverride };
    }
    return (
      <div style={validationStyle}>
        <AlertIcon style={validationErrorStyle.icon} />
        <span
          style={validationErrorStyle.field}
        >
          {' '}
          {validationMessage}
          {' '}
        </span>
      </div>
    );
  }
}

ValidationPlaceholder.propTypes = {
  validationMessage: PropTypes.string,
};

ValidationPlaceholder.defaultProps = {
  validationMessage: '',
};
