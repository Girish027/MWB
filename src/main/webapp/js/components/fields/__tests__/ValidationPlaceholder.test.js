import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidationPlaceholder from 'components/fields/ValidationPlaceholder';

describe('<ValidationPlaceholder />', () => {
  let wrapper;

  const props = {
    validationMessage: 'Name should be greater than 4 char',
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = shallow(<ValidationPlaceholder
        {...props}
      />);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly for default props', () => {
      wrapper = shallow(<ValidationPlaceholder
        {...props}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
