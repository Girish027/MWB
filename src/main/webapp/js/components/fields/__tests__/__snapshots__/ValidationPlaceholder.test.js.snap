// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidationPlaceholder /> Snapshots renders correctly for default props 1`] = `
<div
  style={
    Object {
      "field": Object {
        "color": "red",
        "fontSize": "13px",
      },
      "icon": Object {
        "height": "10px",
      },
      "marginBottom": "10px",
      "marginTop": "-10px",
    }
  }
>
  <AlertIcon
    exclamationMarkColor="#FFF"
    fill="#FF3C27"
    height="1em"
    style={
      Object {
        "height": "10px",
      }
    }
    width="1em"
  />
  <span
    style={
      Object {
        "color": "red",
        "fontSize": "13px",
      }
    }
  >
     
    Name should be greater than 4 char
     
  </span>
</div>
`;
